### PR TITLE
Gptme

### PR DESCRIPTION
**Pull request by 0xbrayo**
> [!IMPORTANT]
> Adds GitHub Actions workflow for bot and refactors Tauri app with file watcher, tray icon sync, and logging.
> 
>   - **GitHub Actions**:
>     - Adds `gptme.yml` workflow to trigger on issue comments using `ErikBjare/gptme` action.
>     - Uses `OPENAI_API_KEY` and `GITHUB_TOKEN` secrets for authentication.
>     - Restricts bot actions to `erikbjare` and `0xbrayo`.
>   - **Tauri Application**:
>     - Adds `SpecificFileWatcher` struct in `lib.rs` to monitor file changes.
>     - Refactors tray icon initialization with `TRAY_CONDVAR` for synchronization.
>     - Introduces `UserConfig` struct for managing user settings.
>     - Initializes logging in `run()` function.
>     - Removes `lib.rs.orig` and `main.rs.orig` files.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ActivityWatch%2Faw-tauri&utm_source=github&utm_medium=referral)<sup> for 86ba1a6604018f3f3fb29a76665a08fc9119711d. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: .github/workflows/gptme.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: run gptme-bot action
+        uses: ErikBjare/gptme/.github/actions/bot@master
+        with:
+          openai_api_key: ${{ secrets.OPENAI_API_KEY }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          allowlist: "erikbjare 0xbrayo"
```
Comment (ErikBjare) [inline .github/workflows/gptme.yml:21]: I think these should maybe be comma separated? (could be wrong)

File: .github/workflows/gptme.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: gptme-bot
+
+on:
+  issue_comment:
+    types: [created]
+
+permissions: write-all
```
[the remark this answers is not in the document]
Comment (ErikBjare) [reply .github/workflows/gptme.yml:7]: It's fine for now, we have the allowlist.

**Comment by ErikBjare** [comment]
Would be cool if you could get this to work, although I fear it might be extra rough around the edges in repos other than gptme.

Might also want to throw in a `gptme.toml` file.

File: .github/workflows/gptme.yml (hunk 1 above)
Comment (0xbrayo) [reply .github/workflows/gptme.yml:21]: Works this way too. But I'll use the comma separation just to be safe

**Comment by 0xbrayo** [comment]
Merged will add a `gptme.toml` later

**Pull request merged**